### ISSUE TITLE
Error generated when dropout * units < 1 but > 0

### ISSUE THREAD

**Issue by eafpres** [rstudio/keras#402]
I use nested loops to parameterize my model in a dense neural net model in keras in RStudio.  The following error is generated.  The error occurs when the product of the dropout factor and the number of units is less than 1 but not identically 0.  When dropouts <- c(0, 0, 0, 0, 0, 0, 0, 0, 0, 0) the error is not generated. 

Error in py_call_impl(callable, dots$args, dots$keywords) : Expecting a string vector: [type=pairlist; required=STRSXP]

units_used_1 <- c(15, 15, 15, 1, 1, 1, 1, 1, 1, 1)
units_used <- rbind(units_used_1,
                    units_used_2,
                    units_used_3)
dropouts_2 <- c(0.026, 0.026, 0.026, 0, 0, 0, 0, 0, 0, 0)
dropouts <- rbind(dropouts_1,
                  dropouts_2,
                  dropouts_3)
...
                  layer_dropout(rate = 
                                    dropouts[dropout_index, 2]} %>%
                  layer_dense(units = 
                                  units_used[unit_index, 2], 
                              activation = activations[3]) %>%`
...
As a suggested enhancement catching this case and not generating a hard error (maybe a warning?) would be appreciated.
